contracts/foundry-tests/LidoOracle.t.sol

```diff
@@ -0,0 +1,106 @@
+
+contract LidoOracleTest is Test, TestConstants, AccessControl {
+    WETH9Mock public weth;
+    ERC20Mock public steth;
+    USDCMock public usdc;
+    LidoOracle public lidoOracle;
+    WstETHMock public lidoMock;
+    ChainlinkMultiOracle public chainlinkMultiOracle;
+    CompositeMultiOracle public compositeMultiOracle;
+    ChainlinkAggregatorV3Mock public stethEthAggregator;
+    ChainlinkAggregatorV3Mock public usdcEthAggregator;
+
+    bytes6 public mockBytes6 = 0xd1eaa762fae7;
+
+    function setUp() public {
+        lidoMock = new WstETHMock();
+        lidoMock.set(1008339308050006006);
+        weth = new WETH9Mock();
+        usdc = new USDCMock();
+        steth = new ERC20Mock("Liquid staked Ether 2.0", "STETH");
+        chainlinkMultiOracle = new ChainlinkMultiOracle();
+        chainlinkMultiOracle.grantRole(0xef532f2e, address(this));
+        stethEthAggregator = new ChainlinkAggregatorV3Mock();
+        usdcEthAggregator = new ChainlinkAggregatorV3Mock();
+        chainlinkMultiOracle.setSource(STETH, steth, ETH, weth, address(stethEthAggregator));
+        chainlinkMultiOracle.setSource(USDC, usdc, ETH, weth, address(usdcEthAggregator));
+        vm.warp(uint256(bytes32(mockBytes6)));
+        stethEthAggregator.set(992415619690099500);
+        usdcEthAggregator.set(WAD / 4000);
+        lidoOracle = new LidoOracle(WSTETH, STETH);
+        lidoOracle.grantRole(0xa8026912, address(this));
+        lidoOracle.setSource(IWstETH(address(lidoMock)));
+        compositeMultiOracle = new CompositeMultiOracle();
+        bytes4[] memory roles = new bytes4[](2);
+        roles[0] = 0x92b45d9c;
+        roles[1] = 0x60509e5f;
+        compositeMultiOracle.grantRoles(roles, address(this));
+        compositeMultiOracle.setSource(WSTETH, STETH, IOracle(address(lidoOracle)));
+        compositeMultiOracle.setSource(STETH, ETH, IOracle(address(chainlinkMultiOracle)));
+        compositeMultiOracle.setSource(USDC, ETH, IOracle(address(chainlinkMultiOracle)));
+        bytes6[] memory path = new bytes6[](1);
+        path[0] = STETH;
+        compositeMultiOracle.setPath(WSTETH, ETH, path);
+        bytes6[] memory paths = new bytes6[](2);
+        paths[0] = STETH;
+        paths[1] = ETH;
+        compositeMultiOracle.setPath(WSTETH, USDC, paths);
+    }
+
+    function testGetConversion() public {
+        (uint256 stethWstethConversion,) = lidoOracle.get(STETH, WSTETH, WAD);
+        require(stethWstethConversion == 991729660855795538);
+        (uint256 wstethStethConversion,) = lidoOracle.get(WSTETH, STETH, 1e18);
+        require(wstethStethConversion == 1008339308050006006);
+    }
+
+    function testRevertOnUnknownSource() public {
+        vm.expectRevert("Source not found");
+        lidoOracle.get(bytes32(DAI), bytes32(mockBytes6), WAD);
+    }
+
+    function testRetrieveDirectPairConversion() public view {
+        (uint256 wstethStethConversion,) = compositeMultiOracle.peek(WSTETH, STETH, 1e18);
```

Comment:
I find it a lot clearer to use ticker+`Amount` for variables.

```
uint256 wstEthAmount = 1e18;
(uint256 stEthAmount,) = compositeMultiOracle.peek(WSTETH, STETH, wstEthAmount);
```

Same in all subsequent tests.